dotCMS/src/main/java/com/dotcms/rest/api/v1/folder/FolderHelper.java

```diff
@@ -152,4 +153,93 @@ private final FolderView getFolders(final Folder folder, final User user){
         return new FolderView(folder,foldersChildCustoms);
     }
 
+    /**
+     * This method will return a list of {@link FolderSearchResultView} that lives under
+     * the path or starts with the path sent.
+     *
+     * @param siteId site where the  folders should live under. If is not send will look under all hosts.
+     * @param pathToSearch path + folder name to find to
+     * @param user user
+     * @return list of {@link FolderSearchResultView}
+     */
+    public List<FolderSearchResultView> findSubFoldersPathByParentPath(final String siteId, final String pathToSearch, final User user)
+            throws DotSecurityException, DotDataException {
+        final List<FolderSearchResultView> subFolders = new ArrayList<>();
+
+        if(pathToSearch.lastIndexOf(StringPool.FORWARD_SLASH) == 0){ //If there is only one / we need to search the subfolders under the host(s)
+            if(UtilMethods.isSet(siteId)) {
+                subFolders.addAll(findSubfoldersUnderHost(siteId,pathToSearch,user));
+            } else{
+                final List<Host> siteList = APILocator.getHostAPI().findAll(user,false);
+                for(final Host site : siteList){
+                    if(subFolders.size() < 20) {
+                        subFolders.addAll(findSubfoldersUnderHost(site.getIdentifier(), pathToSearch, user));
+                    }
+                    continue;
+                }
+            }
+
+        } else {
+            if(UtilMethods.isSet(siteId)) {
```

Comment:
I do not see the diff between the if or the else, take a look

Reply:
one is to search folders under the host (Root level), the other is to search folders under folders.